operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/AbstractDependentResource.java

```diff
@@ -0,0 +1,31 @@
+package io.javaoperatorsdk.operator.api.reconciler.dependent;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+public abstract class AbstractDependentResource<R, P extends HasMetadata>
```

Comment:
If it stays as a Factory (i.e. not exist = create)  it should be renamed. And _sharing_ version could be added (i.e. not exist = update status with missing target condition). The _sharing_ version is another beast because, delete is about undoing what reconcile does not deleting. Again thinks of https://github.com/redhat-developer/service-binding-operator. When the CR is removed, the deployment is updated to suppress any env var that has been injected.
I don't if those two behaviors can be merge in one class